File: week03/week03-5/week03-5.cpp
/// 459.Repeated Substring Pattern
/// week03-5.cpp 學習計畫第5題
///此字串是由重複的字串組合而成
class Solution {
public:
    bool repeatedSubstringPattern(string s) {
        string s2 = s + s;
        int N = s2.length();
        return s2.substr(1, N-2).find(s) != string::npos; //沒有這個位置，很大很大
    }
};

File: week03/week03-5/week03-5.py
# week03-5.py
# 459.Repeated Substring Pattern
class Solution:
    def repeatedSubstringPattern(self, s: str) -> bool:
        s2 = s + s #把字串 s + s 變兩倍長度
        s2 = s2[1:len(s2)-1] #左少1 右少1
        # print(s2)
        if s in s2: return True
        else: return False
